packages/anasazi/epetra/test/OrthoManager/cxx_gentest.cpp

```diff
@@ -625,19 +625,19 @@ int testProjectAndNormalizeGen(RCP<GenOrthoManager<ST,MV,OP> > OM,
     Array<RCP<const MV> > theX, theY, theMX, theMY;
     RCP<SerialDenseMatrix<int,ST> > B = rcp( new SerialDenseMatrix<int,ST>(sizeS,sizeS) );
     Array<RCP<SerialDenseMatrix<int,ST> > > C;
-    if ( (t && 3) == 0 ) {
+    if ( (t & 3) == 0 ) {
```

Comment:
We've tried to fix this before, and it broke the tests.  Do the tests pass for you?

Reply:
That's one thing I'm sorting out now. I swore I ran this through on my own as a GCC 7.2.0 debug build and they all passed just fine. Now that I look at it closer, apparently I was discounting these tests failing as a false positive from something else I was working on.
I could remove the changes, but then I won't be able to set Werror.

Reply:
Though the way it was, it doesn't look like it would ever enter anything except the `((t && x) == 0)` or `( (t && x) == 1 )` lines. It would never find its way into `( (t && x) == 2 )` or `( (t && x) == 3 )`. Those are the ones the Werrors are complaining about:
```
error: comparison of constant ‘2’ with boolean expression is always false [-Werror=bool-compare]
error: comparison of constant ‘3’ with boolean expression is always false [-Werror=bool-compare]
```
So it seems like the tests are sort of broken to start with, or at very least include extraneous `if` statements. I just don't know enough about what they are doing to make sure they get fixed properly without really digging in.
The other simple solution would be to remove the `==2) {...}` & `==3) {...}` segments entirely because they are never actually run anyway as things currently are.
That just seemed too simple before. So, I was taking a stab at what might have been intended given similar if statements above these.
But does just removing the unused `if` statements seem reasonable? If everything is testing correctly, that actually seems like the correct answer.

Reply:
@ZUUL42 wrote:
> I could remove the changes, but then I won't be able to set Werror.

It would make sense to separate out the `TEUCHOS_ASSERT` changes from the `&& 3` etc. changes.  @hkthorn has seen the latter before and might have some insight.

Reply:
Just to try it out, I created a branch and removed the 8 segments of what appear to be dead code the compiler is complaining about, everything `if ( (t && 3) == 2 ) {...}` and `if ( (t && 3) == 3 ) {...}`. See [commit 3196142](https://github.com/ZUUL42/Trilinos/commit/3196142d82af4c3855f3cb19fa72da7e9b23db0e).
I built it with the GCC 7.2.0 PR tester and everything builds without error and passes the tests. However, since that removes a good chunk of code, I wanted to check with @trilinos/anasazi before I push it to this branch.

Reply:
@ZUUL42 I just e-mailed @hkthorn to ask about this.

Reply:
Since it has been 4 weeks, I'd like to propose a path forward.
The compiler is tagging these expressions always false and thus everything inside is never used.
If I remove these sections, the functionality _should_ remain unchanged.
That will get this PR moving forward and allow me to set Werror for Anasazi.
Then if there is any reason to bring back any of the removed code, that can be done with the Werror flag in place.